### ISSUE TITLE
New Abstraction Layer

### ISSUE THREAD

**Issue by tstepanski** [Marusyk/SmartVillageOnline#31]
Suggesting we move the business logic to a service layer and have that layer communicate with the repositories. Controllers will then call this service layer. Async/Await (https://github.com/Marusyk/SmartVillageOnline/issues/25) can then be added to the service layer.

**Comment by Marusyk** [comment]
Thanks. I think this is a good solution. Could you please help me with it?

**Comment by tstepanski** [comment]
Absolutely, I'm on EST (GMT -5) time by the way. I'll probably work on it around 1900/2000 EST.

Timothy M. Stepanski

On Sep 13, 2015, at 10:03, Roma <notifications@github.com<mailto:notifications@github.com>> wrote:


Thanks. I think this is a good solution. Could you please help me with it?

-

**Comment by Marusyk** [comment]
We need to create a new service layer as show on this picture
![webapi architecture](https://cloud.githubusercontent.com/assets/9221694/11797821/63509946-a2ce-11e5-88c4-825ac021e6cc.PNG)

**Comment by Marusyk** [comment]
New structure of solution:

  SmartVillageOnline
   - SV.Domain (All entity framework stuff and all poco classes and interfaces)
   - SV.UnitTests (All unittests for all projects)
   - SV.Service (All business logic)
   - SV.WebUI (All font end logic)

**Comment by Marusyk** [comment]
Added a new class library to the solution named SV.Service. This layer will act as our business logic layer. Note that, we can make use of our API controllers to write business logic, but I am trying to segregate my business logic in an extra layer so that if in future I want to use WCF, MVC, ASP.NET Web Pages or any other application as my presentation layer then I can easily integrate my Business logic layer in it.
